src/main/java/org/xbill/DNS/LookupResult.java

```diff
@@ -0,0 +1,61 @@
+package org.xbill.DNS;
+
+import static java.util.Objects.hash;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.xbill.DNS.exceptions.AdditionalDetail;
+
+/** LookupResult instances holds the result of a successful lookup operation. */
+public class LookupResult {
+  private final List<Record> records;
+  private final AdditionalDetail additionalDetail;
+
+  /**
+   * Construct an instance with the provided records.
+   *
+   * @param records a list of records to return, or null if there was no response
+   * @param additionalDetail additional detail on this response, such as the reason for records
+   *     being null.
+   */
+  public LookupResult(List<Record> records, AdditionalDetail additionalDetail) {
+    this.records = Collections.unmodifiableList(new ArrayList<>(records));
+    if (records.isEmpty() && additionalDetail == null) {
+      throw new IllegalArgumentException(
+          "Empty records can not be combined with null additionalDetail");
+    }
+    this.additionalDetail = additionalDetail;
+  }
+
+  /**
+   * An unmodifiable list of records that this instance wraps
+   *
+   * @return an unmodifiable List of Record instances.
+   */
+  public List<Record> get() {
+    return records;
+  }
+
+  public AdditionalDetail getAdditionalDetail() {
+    return additionalDetail;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    LookupResult that = (LookupResult) o;
+    return records.equals(that.records) && additionalDetail == that.additionalDetail;
+  }
+
+  @Override
+  public int hashCode() {
+    return hash(records, additionalDetail);
+  }
+
+  @Override
+  public String toString() {
```

Comment:
Can also be done with [Lombok](https://projectlombok.org/features/ToString) `@ToString`.

Reply:
changing